File: BestTimeToBuySellStock.java
//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int best = 0;
        int potential = 0;

        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < min) {
                min = prices[i];
            }
            potential = prices[i] - min;
            if (potential > best) {
                best = potential;
            }
        }
        return best;
    }
}

File: Easy/BestTimeToBuySellStock/BestTimeToBuySellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result = 0;
        int min = prices[0];
        int potential = 0;
        
        for(int i = 1; i < prices.size(); i++){
            if(prices[i] < min){
                min = prices[i];
            }
            potential = prices[i]-min;
            if(potential>result){
                result = potential;
            }
        }
        return result;
    }
};